sdk_extensions/logging/support/src/main/java/io/opentelemetry/common/AnyValue.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import com.google.auto.value.AutoValue;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A class that represents all the possible values for a data body. An {@code AnyValue} can have 6
+ * types of values: {@code String}, {@code boolean}, {@code int}, {@code double}, {@code array}, or
+ * {@code kvlist}. represented through {@code AnyValue.Type}. A {@code array} or a {@code kvlist}
+ * can in turn hold other {@code AnyValue} instances, allowing for mapping to JSON-like structures.
+ *
+ * @since 0.8.0
+ */
+@Immutable
+public abstract class AnyValue {
+
+  /** An enum that represents all the possible value types for an {@code AnyValue}. */
+  public enum Type {
+    STRING,
+    BOOL,
+    INT,
+    DOUBLE,
+    ARRAY,
+    KVLIST
+  }
+
+  /**
+   * Returns an {@code AnyValue} with a string value.
+   *
+   * @param stringValue The new value.
+   * @return an {@code AnyValue} with a string value.
+   */
+  public static AnyValue stringAnyValue(String stringValue) {
+    return AnyValueString.create(stringValue);
+  }
+
+  /**
+   * Returns the string value of this {@code AnyValue}. An UnsupportedOperationException will be
+   * thrown if getType() is not {@link AnyValue.Type#STRING}.
+   *
+   * @return the string value of this {@code AttributeValue}.
+   */
+  public String getStringValue() {
+    throw new UnsupportedOperationException(
+        String.format("This type can only return %s data", getType().name()));
+  }
+
+  /**
+   * Returns an {@code AnyValue} with an int value.
+   *
+   * @param intValue The new value.
+   * @return an {@code AnyValue} with a int value.
+   */
+  public static AnyValue intAnyValue(int intValue) {
```

Comment:
I assume AnyValue here mirrors AnyValue from OTLP Protobufs. If that's the case then we likely need to use `long` instead of `int` since OTLP Protobufs use int64.